### PR TITLE
[sonic-sdk] add sonic sdk and sonic sdk buildenv

### PR DESCRIPTION
**Pull request by stepanblyschak**
**- Why I did it**

To give SONiC Application Extension developers an environment to run and develop their apps.

**- How I did it**
Created sonic-sdk and sonic-sdk-buildenv dockers and their dbg versions.

**- How to verify it**
Build:

```
$ make -f slave target/sonic-sdk.gz target/sonic-sdk-buildenv.gz
```

**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by liat-grozovik** [comment]
@renukamanavalan and @lguohan this is a change which IMO we can take to master even if we didn't close the flow on building it and where to share and provide the artifacts. Can you please review?
BTW this is already being used internally and is very helpful while developing app extensions.

**Comment by liat-grozovik** [comment]
@lguohan any way to rerun the KVM test? it is not restarted

**Comment by stepanblyschak** [comment]
@bandaru-viswanath Could you please review?

File: platform/vs/sonic-version/sonic_version.yml.j2 [hunk 1]
```diff
@@ -0,0 +1 @@
+../../../files/build_templates/sonic_version.yml.j2
```
Comment (renukamanavalan) [inline platform/vs/sonic-version/sonic_version.yml.j2:1]: Is there not a pre-existing env variable that defines the path ?

File: rules/docker-config-engine-buster.mk [hunk 2]
```diff
@@ -11,7 +11,9 @@ $(DOCKER_CONFIG_ENGINE_BUSTER)_LOAD_DOCKERS += $(DOCKER_BASE_BUSTER)
 $(DOCKER_CONFIG_ENGINE_BUSTER)_FILES += $(SWSS_VARS_TEMPLATE)
 $(DOCKER_CONFIG_ENGINE_BUSTER)_FILES += $($(SONIC_CTRMGRD)_CONTAINER_SCRIPT)
 
-$(DOCKER_CONFIG_ENGINE_BUSTER)_DBG_DEPENDS = $($(DOCKER_BASE_BUSTER)_DBG_DEPENDS)
+$(DOCKER_CONFIG_ENGINE_BUSTER)_DBG_DEPENDS = $($(DOCKER_BASE_BUSTER)_DBG_DEPENDS) \
+                                             $(LIBSWSSCOMMON_DBG) \
```
Comment (renukamanavalan) [inline rules/docker-config-engine-buster.mk:15]: What demands this new dependency?

File: rules/docker-sonic-sdk-buildenv.mk [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# SONiC SDK Docker Image
+
```
Comment (renukamanavalan) [inline rules/docker-sonic-sdk-buildenv.mk:2]: Why there are two SD dockers ?
sonic-sdk
sonic-sdk-buildenv

What are their use cases?

File: rules/docker-config-engine-buster.mk (hunk 2 above)
Comment (stepanblyschak) [reply rules/docker-config-engine-buster.mk:15]: docker-config-engine has libswsscommon, but docker-config-engine-dbg does not have libswsscommon-dbg nor libhiredis-dbg

File: platform/vs/sonic-version/sonic_version.yml.j2 (hunk 1 above)
Comment (stepanblyschak) [reply platform/vs/sonic-version/sonic_version.yml.j2:1]: This is a symbolic link or are you refering to https://github.com/Azure/sonic-buildimage/pull/6712/files/f11d60e07ba8cbee72a1aa9f83b4db70f1701b17#diff-b815aaa1a414e83000682ee2995e261dc33c72c4a2ff21771dc8ed1246416e53R485?
If yes, at build_debian.sh stage of build there is no BUILD_TEMPLATES environment variable that defines this path.

**Comment by liat-grozovik** [comment]
@renukamanavalan now that tests is passing do we have any further open issues with this PR or we can move forward and merge it?

**Review by renukamanavalan — APPROVED** [review]

**Pull request merged**